EXPORT MODEL (#191)
nils-braun/dask-sql: docs/pages/machine_learning.rst

using the given name.
Have a look into :ref:`ml` for more information.

4. Check Model parameters - Model meta data
-------------------------------------------

> After the Model was trained, you can inspect and get model details by using following

Reviewer: ```suggestion
After the model was trained, you can inspect and get model details by using the following
```

Author: updated 👍

sql statements

.. code-block:: sql

>     -- show list of models  which are trained and saved in the context.

Reviewer: ```suggestion
    -- show the list of models which are trained and stored in the context.
```

    SHOW MODELS

>     -- To get Hyperparameters of the trained MODEL use ,

Reviewer: ```suggestion
    -- To get the hyperparameters of the trained model, use 
```

    -- DESCRIBE MODEL <model_name>.
    DESCRIBE MODEL my_model

5. Export Trained Model
------------------------
Once your model was trained and performs good in your validation dataset,

> you can export the model into a file with supporting model serilization

Reviewer: ```suggestion
you can export the model into a file with one of the supported model serialization
```

> format like pickle,joblib,mlflow (framework agnostic serilization format) etc.

Reviewer: ```suggestion
formats like pickle, joblib, mlflow (framework-agnostic serialization format), etc.
```

> Currently dask_sql supports pickle,joblib and mlflow
> formats for exporting the trained model which can be deployed as microservices etc

Reviewer: ```suggestion
Currently, dask-sql supports the pickle, joblib and mlflow format for exporting the trained model, which can then be deployed as microservices etc
```

Before training and exporting the models from different framework like
lightgbm, catboost, please ensure the relevant packages are installed in the

> dask_sql environment otherwise it will raises RuntimeError and if you

Reviewer: ```suggestion
dask-sql environment, otherwise it will raise an exception on import and if you
```

> are using mlflow as model_serde ensure mlflow was installed

Reviewer: If you change the name of the parameter, also change it here :-)

Author: updated 👍

> .. code-block:: sql

Reviewer: Just a matter of taste, but I like the closing ")" on the next line better (as you have done further down), so:
```sql
EXPORT MODEL (
    ...
)
```

Author: yeah it's really looking nice this way, changed in most of the places. 👍

> 
> 

Reviewer: ```suggestion

```

>     -- for pickle model serilization

Reviewer: ```suggestion
    -- for pickle model serialization
```

    EXPORT MODEL my_model WITH (
        model_serde ='pickle',
        location = 'model.pkl')

>     -- for joblib model serilization

Reviewer: ```suggestion
    -- for joblib model serialization
```

    EXPORT MODEL my_model WITH (
        model_serde ='joblib',
        location = 'model.pkl')

>     -- for mlflow model serilization

Reviewer: ```suggestion
    -- for mlflow model serialization
```

    EXPORT MODEL my_model WITH (
        model_serde ='mlflow',
        location = 'mlflow_dir')

    -- Note you can pass more number of key value pairs

>     -- (parameters) which will delegated to respective

Reviewer: ```suggestion
    -- (parameters) which will be delegated to the respective
```
